exasol/error/_cli.py

```diff
@@ -0,0 +1,354 @@
+                )
+            )
+
+        # Validate that everything except parameters are constants
+        if isinstance(mitigations, str) and not isinstance(mitigations, ast.Constant):
+            errors.append(
+                Validation.Error(
+                    message=msg.format(type="mitigations", value=mitigations.value),
+                    file=file,
+                    line_number=mitigations.lineno,
+                )
+            )
+
+        if isinstance(mitigations, list):
+            invalid = [e for e in mitigations.elts if not isinstance(e, ast.Constant)]
+            errors.extend(
+                [
+                    Validation.Error(
+                        message=msg.format(type="mitigations", value=e.value),
+                        file=file,
+                        line_number=e.lineno,
+                    )
+                    for e in invalid
+                ]
+            )
+        # Validate parameters
+        for key in parameters.keys:
+            if not isinstance(key, ast.Constant):
+                errors.append(
+                    Validation.Error(
+                        message=msg.format(type="key", value=key.value),
+                        file=file,
+                        line_number=key.lineno,
+                    )
+                )
+        for value in parameters.values:
+            if isinstance(value, ast.Call):
+                description = value.args[1]
+                if not isinstance(description, ast.Constant):
+                    errors.append(
+                        Validation.Error(
+                            message=msg.format(
+                                type="description", value=description.value
+                            ),
+                            file=file,
+                            line_number=value.lineno,
+                        )
+                    )
+
+        return errors, warnings
+
+    @staticmethod
+    def _is_exa_error(node):
+        if not isinstance(node, ast.Call):
+            return False
+        name = getattr(node.func, "id", "")
+        name = getattr(node.func, "attr", "") if name == "" else name
+        return name == "ExaError"
+
+    def _make_error(self, node):
+        code, message, mitigations, parameters = node.args
+
+        def normalize(params):
+            for k, v in zip(params.keys, params.keys):
+                if isinstance(v, ast.Call):
+                    yield k.value, v[1]
+                else:
+                    yield k.value, ""
+
+        return ErrorCodeDetails(
+            identifier=code.value,
+            message=message.value,
+            messagePlaceholders=[
+                Placeholder(name, description)
+                for name, description in normalize(parameters)
+            ],
+            description=None,
+            internalDescription=None,
+            potentialCauses=None,
+            mitigations=[m.value for m in mitigations.elts],
```

Comment:
` mitigations.elts` ?

Reply:
This is the member of  `ast.List` and `ast.Tuple` holding the elements.

Reply:
Ah, understood